23andMe/Yamale
# feature/HEL-3992 Add kwarg for none

The problem:
If you have an optional node and you need it to have a value, there is no way to enforce this.

The solution:
Added a new kwarg called none. If none is true, a value is not required. If none is false, a value is required. By default, none is true for backwards compatibility.

A note:
I was initially going to add this as a constraint, but this was a problem because the code first checks if an optional node has no value. If no value is detected, constraints are not processed for the node. The solution would have required more code changes and just seemed a bit less elegant to me.

Take a close look at those tests! If you need more, let me know.

To test with product:
1. For product, load feature/HEL-3992
2. For yamale, load feature/HEL-3992
3. In yamale, follow these steps to create a local distribution of the yamale package and to load that package into product: 
https://23andme.atlassian.net/wiki/spaces/~111242565/pages/558170249/WIP+PIP+package+and+Vagrant
4. In product, go to data23/health/assessment/lark.
5. Add a new field to the schema. It could look like this:
mynode: str(required=False, none=False)
5. Add a new field to one of the other yamls in this directory that does not have a value. It could look like this:
mynode:
6. In product, run: (cd data23/health && make clean && make)
You should see an error pop up for mynode.
